### PR TITLE
ReferenceRequestDetailAPI

### PR DESCRIPTION
**Pull request by ChrisCurrin**
- add repo for applicationModel to get form, sections, question
- add tests for repo for applicationModel
- update repo for response to include 
  - get response by user_id
  - get answers by response_id
  - get answers by question_id 
  - get answer by response_id and by question_id
  - getting [(Question, Answer)] by section_id and response_id 
- added tests for updated response repo
- reverted response model from previous commit to adjust PR 624 based on feedback (exclude candidate_title, etc.) 
- update testing to add multiple dummy users at once from a list of names

### CODE REVIEW

**Comment by ChrisCurrin** [comment]
tests were working locally, will check them out next week (going away for the weekend) and revert back

File: api/app/responses/tests.py [hunk 1]
```diff
@@ -45,43 +57,66 @@ def _seed_data(self):
...
+        # responses
         self.test_response = Response(
-            self.test_form.id, self.other_user_data['id'])
+                self.test_form.id, self.other_user_data['id'])
         _add_object_to_db(self.test_response)
 
         self.test_answer1 = Answer(
-            self.test_response.id, self.test_question.id, 'My Answer')
+                self.test_response.id, self.test_question.id, 'My Answer')
         _add_object_to_db(self.test_answer1)
```
Comment (avishkar58) [inline api/app/responses/tests.py]: Can you update these to use your self.add_to_db function instead and remove the _add_object_to_db function

File: api/app/utils/testing.py [hunk 2]
```diff
@@ -19,13 +24,51 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
...
+    :param text: The input string.
+    :type text: String.
+
+    :returns: The processed String.
+    :rtype: String.
+    """
+    try:
+        text = unicode(text, 'utf-8')
+    except (TypeError, NameError): # unicode is a default on python 3
+        pass
```
Comment (avishkar58) [inline api/app/utils/testing.py]: `six.ensure_text` ?

**Comment by avishkar58** [comment]
Hey @ChrisCurrin - did you try test this locally using docker? Was there some step you needed to do to get it to mount the names.txt file? It also fails on my machine with the same error as on CircleCI...

**Comment by avishkar58** [comment]
Updated the path and the tests are working now.

File: api/app/utils/testing.py (hunk 2 above)
Comment (ChrisCurrin) [reply api/app/utils/testing.py]: good suggestion, thanks!

**Comment by ChrisCurrin** [comment]
sorry about the mess with the merges 🙈

> I accidentally merged feature branch into local develop, but "undoing" and reverting didn't fix it, so after a series of merges (and cherry-picking changes), there should be parity with the initial feature implementation and the latest changes to develop

and the delay of course! I'll be faster with addressing comments.

**Pull request merged**